lib/valueflow.cpp

```diff
@@ -3497,6 +3499,17 @@ bool isLifetimeBorrowed(const Token *tok, const Settings *settings)
         if (!Token::simpleMatch(tok, "{")) {
             const ValueType *vt = tok->valueType();
             const ValueType *vtParent = tok->astParent()->valueType();
+            ValueType svt;
+            // TODO: Move logic to ValueType
```

Comment:
Do you plan to do this within this PR or get it merged as is?

Reply:
The plan it to add this in a future PR. We do something similar [here](https://github.com/danmar/cppcheck/blob/master/lib/checknullpointer.cpp#L336) as well, which should be moved to ValueType.